packages/react-native-web/src/exports/Image/index.js

```diff
@@ -332,14 +343,67 @@ const Image: React.AbstractComponent<
   );
 });
 
-Image.displayName = 'Image';
+BaseImage.displayName = 'Image';
+
+/**
+ * This component handles specifically loading an image source with headers
+ * default source is never loaded using headers
+ */
+const ImageWithHeaders: ImageComponent = React.forwardRef((props, ref) => {
+  // $FlowIgnore
+  const nextSource: ImageSource = props.source;
+  const [blobUri, setBlobUri] = React.useState('');
+  const request = React.useRef<LoadRequest>({
+    cancel: () => {},
+    source: { uri: '', headers: {} },
+    promise: Promise.resolve('')
+  });
+
+  const { onError, onLoadStart, onLoadEnd } = props;
+
+  React.useEffect(() => {
+    if (!hasSourceDiff(nextSource, request.current.source)) return;
+
+    // When source changes we want to clean up any old/running requests
+    request.current.cancel();
+
+    if (onLoadStart) onLoadStart();
+
+    request.current = ImageLoader.loadWithHeaders(nextSource);
```

Comment:
I think this line would be a good place to leave a comment that explains [this comment](https://github.com/Expensify/react-native-web/pull/11/files#r1084967401).